package.json

```diff
@@ -61,19 +63,19 @@
     "yaml-front-matter": "^4.0.0"
   },
   "devDependencies": {
-    "babel-core": "^6.26.0",
-    "babel-eslint": "^8.2.3",
+    "@babel/core": "^7.19.0",
```

Comment:
Interesting that 177b0aefa2c327b0c28f677776ed3b787cb99421 which merely updates the Babel core dependency in the top-level project causes `gatsby build` to fail in `static-site`:

```
failed Building production JavaScript and CSS bundles - 63.137s
error Generating JavaScript bundles failed

Can't resolve 'regenerator-runtime/runtime' in '/home/runner/work/nextstrain.org/nextstrain.org/auspice-client/customisations'

Perhaps you need to install the package 'regenerator-runtime/runtime'?
```

Reply:
I don't understand why, but 4a4f1839d9d02f4c5d85c11b5a7309b6a605d0a8 fixed this.

Maybe relevant:

- https://github.com/gatsbyjs/gatsby/issues/23676
- https://github.com/gatsbyjs/gatsby/issues/1905
- https://stackoverflow.com/a/68091315

Reply:
> I don't understand why, but



> Interesting that [177b0ae](https://github.com/nextstrain/nextstrain.org/commit/177b0aefa2c327b0c28f677776ed3b787cb99421) which merely updates the Babel core dependency in the top-level project causes `gatsby build` to fail in `static-site`:

Did you have a populated `static-site/node_modules/` when this happened? If not, Node would have found the one in the repo root and picked up the new Babel version.

Or maybe it's because, when resolving imports, Node will, I believe, keep walking upwards looking for other `node_modules/` if the first one it finds doesn't have the module being imported. So if something in `static-site/node_modules/` was doing a conditional import or similar, the two deps could get tangled up.

Reply:
![image](https://user-images.githubusercontent.com/79913/189245557-66fd1882-4bee-4b2f-bf8c-7f3eb4e1bc8e.png)

Reply:
> Did you have a populated `static-site/node_modules/` when this happened?

Yes, this was reproduced on [the CI run](https://github.com/nextstrain/nextstrain.org/runs/8256127348?check_suite_focus=true) which runs `npm ci` + `build.sh`, which is effectively:

```
npm ci
cd static-site
npm ci
gatsby build
```

Also noting that the path in the error is `nextstrain.org/auspice-client/customisations`. `auspice-client` is only used in one file in `static-site/`. Removing that reference allows the build to succeed: c9ba5f9abdd55dc06822441c01906389b0c0fb38 (not in PR)

Maybe similar to what you described, because there is no `node_modules` under that path, it then walks up to `nextstrain.org/node_modules/` when really `nextstrain.org/static-site/node_modules` would be more appropriate.

Reply:
So maybe the answer to "why did 4a4f1839d9d02f4c5d85c11b5a7309b6a605d0a8 fix this?" is multiple parts:

1. `static-site/src/sections/sars-cov-2-page.jsx` relies on the top-level `node_modules`
2. 177b0aefa2c327b0c28f677776ed3b787cb99421 effectively removed `regenerator-runtime` from the top-level `node_modules`
3. Using the older version of Gatsby to create the page from (1) relies on `regenerator-runtime`. The newer version of Gatsby does not have this dependency.

Reply:
More notes (or partial rememberings) for posterity -- node's "ability" to walk up is why the folder structure when building auspice with custom extensions really matters, and why we put the JS in a subdirectory (`auspice-client/customisations`).

P.S. How have we persevered with Gatsby for this long?!?

Reply:
> P.S. How have we persevered with Gatsby for this long?!?

I think I'm going to rip it out once we start doing more UI work… :see_no_evil:

Reply:
Commit message updated in 383adab98fe1375d772125f2f7117458a4aaf860.